### PR TITLE
Add maintenance script to automate test rebaselining. NFC

### PR DESCRIPTION
**Pull request by sbc100**
This is just the first iteration.

See #23146 for more plans.

### CODE REVIEW

**Comment by sbc100** [comment]
For an example of the PR that this script creates see https://github.com/emscripten-core/emscripten/pull/23163

File: tools/maint/rebaseline_tests.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+The following ({len(filenames)}) test expectation files were updated by
+running the tests with `--rebaseline`:
+
+'''
+
+  for file in filenames:
+    commit_message += process_changed_file(file)
+
+  commit_message += f'\nAverage change: {statistics.mean(all_deltas):+.2f}%\n'
```
Comment (kripken) [inline tools/maint/rebaseline_tests.py:100]:
```suggestion
  commit_message += f'\nAverage change: {statistics.mean(all_deltas):+.2f}% ({min(all_deltas):+.2f}% - {max(all_deltas):+.2f}%)\n'
```

**Review by kripken — APPROVED** [review]

File: tools/maint/rebaseline_tests.py (hunk 1 above)
Comment (kripken) [reply tools/maint/rebaseline_tests.py:100]: (min and max seem important too, as even if the mean looks good, there could be a single big regression)

File: tools/maint/rebaseline_tests.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+  print(f'processing {filename}')
+  if len(content.splitlines()) == 1:
+    size = int(content.strip())
+    old_size = int(old_content.strip())
+  else:
+    try:
+      current_json = json.loads(content)
+      old_json = json.loads(old_content)
+    except Exception:
+      print(f'{filename}: Unable to passe json content. Unsupported file format?')
```
Comment (dschuff) [inline tools/maint/rebaseline_tests.py:52]:
```suggestion
      print(f'{filename}: Unable to parse json content. Unsupported file format?')
```

File: tools/maint/rebaseline_tests.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+  filename = utils.removeprefix(filename, 'test/')
+  delta = size - old_size
+  percent_delta = delta * 100 / old_size
+  all_deltas.append(percent_delta)
+  return f'{filename}: {old_size} => {size} [{delta:+} bytes / {percent_delta:+.2f}%]\n'
+
+
+def main(argv):
+  parser = argparse.ArgumentParser()
+  parser.add_argument('-s', '--skip-tests', action='store_true', help="don't actualy run the tests, just analyize the existing results")
```
Comment (dschuff) [inline tools/maint/rebaseline_tests.py:66]:
```suggestion
  parser.add_argument('-s', '--skip-tests', action='store_true', help="don't actually run the tests, just analyze the existing results")
```

File: tools/maint/rebaseline_tests.py (hunk 1 above)
Comment (sbc100) [reply tools/maint/rebaseline_tests.py:100]: Done

**Pull request merged**